app/models/paper.rb

```diff
@@ -43,4 +45,10 @@ def valid_methodologies
       'case report'
     ]
   end
+
+  def find_publication
+    if self.publication.present?
+      self.publication = Publication.where(name: self.publication.name).first_or_initialize
```

Comment:
So I kinda just did stuff until I got the behavior I was expecting.  I
originally started with a validate_uniqueness in publications.  Problem was
if I tried to save a paper with an existing publication, it would throw an
error.  So I added this before_validation to get it to get the params info
before the validation because the validation would throw away the data
before it got to the controller.  That was still causing an error as the
nested attribute wouldn't let me save with a different id when I was
updating.

So I got rid of the validate_uniqueness and it all seemed to work.  I
figure this method would keep us from creating the same journal multiple
times.  It should I think also work for the Pubmed API because this method
is sitting in the model?  Anyway, we can partner code this as well cuz I
took multiple rounds to try to get this functionality to work but I'm only
60% conscious of all the ramifications.

On Tue, Aug 23, 2016 at 2:51 PM, Brian Kung <notifications@github.com>
wrote:

> In app/models/paper.rb
> <https://github.com/briankung/projectcredo/pull/64#discussion_r75937890>:
>
> > @@ -43,4 +45,10 @@ def valid_methodologies
> >        'case report'
> >      ]
> >    end
> > +
> > +  def find_publication
> > +    if self.publication.present?
> > +      self.publication = Publication.where(name: self.publication.name).first_or_initialize
>
> I'm still not entirely sure why you're doing this, but I think you can do
> this: http://guides.rubyonrails.org/active_record_querying.html#
> find-or-initialize-by
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/briankung/projectcredo/pull/64/files/c10ff7dd636d241cfd3c03b300def8df2400222c#r75937890>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACKsZLmMj7szfS2wkSN3lIWH0LLcqOanks5qi08qgaJpZM4JqR56>
> .
>